cabal.project

```diff
@@ -30,6 +30,7 @@ constraints:
   , bimap >= 0.4.0
   , ListLike >= 4.7.3
   , witherable >= 0.4
+  , Win32 == 2.6.2.0
```

Comment:
This seems suspect to me, but that's mostly because I'm ignorant re windows deps.

@coot, what do you think?

Reply:
It does not sounds right, why `Win32 >= 2.6.2.0` (which is already declared in `ouroboros-consensus.cabal`), does not work?

Reply:
If I use `Win32 >= 2.6.2.0`, it automatically picks `Win32-2.10.1.1` (not sure why it doesn't jump all the way up to `Win32-2.12.0.1`), but with `2.10.1.1` haskell.nix seems to have some problems to build it:
```
❯ nix build -f release.nix x86_64-w64-mingw32.tests.ouroboros-consensus-cardano-test.test.x86_64-linux
...
[1/0/325 built] building Win32-lib-Win32-2.10.1.1-x86_64-w64-mingw32 (buildPhase): Preprocessing library for Win32-2.10.1.1..
```
It seems to deadlock at the preprocessing step for some reason that escapes my knowledge of Nix. Master (as seen [here](https://hydra.iohk.io/build/7046919#tabs-build-deps)) uses Win32-2.8.5.0, which also makes haskell.nix deadlock in the preprocessing step:
```
❯ nix build -f release.nix x86_64-w64-mingw32.tests.ouroboros-consensus-cardano-test.test.x86_64-linux
...
[1/5/330 built, 1 copied (0.1/0.1 MiB), 0.2 MiB DL] building Win32-lib-Win32-2.8.5.0-x86_64-w64-mingw32 (buildPhase): Preprocessing library for Win32-2.8.5.0..
```

I understand this is not acceptable and I'm not planning on merging this until we have a solution for Win32.

Reply:
The compilation doesn't deadlock, just takes a lot of time. When doing this in Hydra, it progresses further and then fails when compiling io-sim library:
```
Preprocessing library for io-sim-0.2.0.0..
Building library for io-sim-0.2.0.0..
[1 of 2] Compiling Control.Monad.IOSim.Internal ( src/Control/Monad/IOSim/Internal.hs, dist/build/Control/Monad/IOSim/Internal.o )
---> Starting remote-iserv on port 7515
---| remote-iserv should have started on 7515
Could not load wine-gecko. HTML rendering will be disabled.
wine: configuration in '/build' has been updated.
Listening on port 7515
remote-iserv.exe:  | /nix/store/fyqacymzm7d7py7n8vw9y51s6zj9l2zk-Win32-lib-Win32-2.10.1.1-x86_64-w64-mingw32/lib/x86_64-windows-ghc-8.10.4/Win32-2.10.1.1-KzpEiQBEAFE8GpgOOc9GS6/HSWin32-2.10.1.1-KzpEiQBEAFE8GpgOOc9GS6.o: unknown symbol `GetNLSVersionEx'
remote-iserv.exe: Could not load Object Code /nix/store/fyqacymzm7d7py7n8vw9y51s6zj9l2zk-Win32-lib-Win32-2.10.1.1-x86_64-w64-mingw32/lib/x86_64-windows-ghc-8.10.4/Win32-2.10.1.1-KzpEiQBEAFE8GpgOOc9GS6/HSWin32-2.10.1.1-KzpEiQBEAFE8GpgOOc9GS6.o.


<no location info>: error:
    ghc: unable to load package `Win32-2.10.1.1'
---> killing remote-iserv...
```

Reply:
If #3286 gets merged, we will not have this problem anymore as the version of Win32 will be fixed to 2.6.1.0. The problem of building later versions in nix will still be present but it will be non-urgent then.

Reply:
#3286 is merged so we are now using 2.6.1.0.